src/async/timing.ts

```diff
@@ -4,13 +4,13 @@ import Promise from 'dojo-shim/Promise';
  * Used for delaying a Promise chain for a specific number of milliseconds.
  *
  * @param milliseconds the number of milliseconds to delay
- * @return {function(T): Promise<T>} a function producing a promise that eventually returns the value passed to it; usable with Thenable.then()
+ * @return {function(T): Promise<T>} a function producing a promise that eventually returns the value or execute the value function passed to it; usable with Thenable.then()
  */
 export function delay<T>(milliseconds: number): Identity<T> {
 	return function (value: T): Promise<T> {
```

Comment:
I agree with unwrapping the `thenable`. However I think there might be a use case for it:
```
			const delayed = timing.delay(250);
			delayed('foo');
			delayed(function() { return 'bar'; });
```
Calling `delay` with a time but without a value, can sort of be used as a partial function. And the user can then apply that function multiple times without recreating the delay.  If it doesn't make sense, I can remove the `thenable` wrap.